### PR TITLE
[W5.7][W10-4]Yuen Jun Rong

### PR DESCRIPTION
**Pull request by junrong09**
Ready for review

### CODE REVIEW

**Review by barkhyehyeon — COMMENTED** [review]
For the sort method, I think it would be better to use toString() method of Name object rather than calling fullName field directly. (Although I am not sure why fullName is set as public while its get method - toString() exists. In perspective of OOP, I think it is more appropriate to set fullName field as private.)

**Comment by BennTay** [comment]
In Parser.java line 92 and 93, I think you might have overlooked that the lines are improperly indented according to the style guide.

**Review by BennTay — COMMENTED** [review]
In Parser.java line 92 and 93, I think you might have overlooked that the lines are improperly indented according to the style guide.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Sort all persons : `sort`
+
+Sort all persons in the address book by name in lexical order. +
```
Comment (AceCentury) [inline docs/UserGuide.adoc:77]: Good that you indicated how the persons are sorted.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -129,6 +129,12 @@ public void clear() {
         internalList.clear();
     }
 
+    /**
+     * Sort all persons in alphabetical (ignoring case) order.
+     */
+    public void sort() { internalList.sort((x,y)
+            -> x.getName().fullName.compareToIgnoreCase(y.getName().fullName));}
```
Comment (AceCentury) [inline src/seedu/addressbook/data/person/UniquePersonList.java:136]: Might want to consider breaking before the dot (.) operator as per the coding standard to improve readability.

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -88,6 +89,9 @@ public Command parseCommand(String userInput) {
         case ListCommand.COMMAND_WORD:
             return new ListCommand();
 
+            case SortCommand.COMMAND_WORD:
```
Comment (AceCentury) [inline src/seedu/addressbook/parser/Parser.java:92]: Minor indentation issue.

File: test/input.txt [hunk 4]
```diff
@@ -54,6 +54,23 @@
...
+  sort
+  list
+
+  add Aann Tan p/12341234 e/aann@mail.com a/123, Royal Road
+  list
+  sort
+  list
+
+  delete 1
+  list
```
Comment (AceCentury) [inline test/input.txt:72]: Quite a comprehensive system test!

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package seedu.addressbook.commands;
+
+import org.junit.Before;
+import org.junit.Test;
```
Comment (AceCentury) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:4]: Ideally there should be a line break after this import statement as they are separate groups.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TestUtil;
```
Comment (AceCentury) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:12]: Overall good job on following the coding standard for import statements!

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
```
Comment (AceCentury) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:17]: Please avoid wildcard imports as per coding standards. You were doing so well! :(

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+    private AddressBook emptyAddressBook;
+    private AddressBook addressBook;
+    private List<ReadOnlyPerson> emptyDisplayList;
+    private List<ReadOnlyPerson> listWithEveryone;
+    private SortCommand sortCommand;
+
+    @Before
+    public void setUp() throws Exception {
+        Person johnDoe = new Person(new Name("John Doe"), new Phone("61234567", false),
```
Comment (AceCentury) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:29]: Might want to consider having line breaks after every comma (as per coding standard) and also having a line break between each person for readability.

**Review by AceCentury — APPROVED** [review]
Please close this PR after reading comments.

**Pull request closed without merging**